### PR TITLE
feat(glossary): adding materialized view to glossary

### PR DESCRIPTION
**Pull request by amyshwang**

### CODE REVIEW

File: src/pages/glossary/materialized-view.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/>
+
+## Materialized views vs. traditional views
+
+Materialized views differ from traditional views in that they are physically
+stored in the database. Traditional views are virtual in nature and executed on
+the fly when they are accessed. On the other hand, materialized views are
+pre-computed and physically stored. Therefore, when querying against the materialized
+view, the database does not need to re-run the associated query every time.
+Essentially, materialized view trades storage for performance.
```
Comment (jerrinot) [inline src/pages/glossary/materialized-view.md:34]: maybe we should be more explicit:
> Essentially, materialized view trades storage for query performance.
Query performance is what is improved. However, depending on the implementation, ingestion performance may get worse.

**Review by jerrinot — APPROVED** [review]
a minor clarification then good to merge.

you could perhaps think whether/how to link the "stream processing" article. there is an overlap in use-cases between auto-updated materialized views and stream processing.

File: src/pages/glossary/materialized-view.md [hunk 2]
```diff
@@ -51,7 +51,8 @@ database implementation:
   the refresh command. This is useful when the view does not need to update
   frequently.
 - **Automatic refresh**: Materialized views can also be refreshed automatically
-  via some trigger or predefined schedule.
+  based on some triggers or a predefined schedule. This is made possible by
+  resorting to stateful [stream processing](/glossary/stream-processing/).
```
Comment (jerrinot) [inline src/pages/glossary/materialized-view.md:55]: I'm afraid I confused you:(

> This is made possible by resorting to stateful

I think this sentence creates confusion. it's not right to say "resorts to stateful stream processing". What I meant to say: There is a similarity between stateful stream processing and materialized views. This similarity is non-obvious. Historically "stream processors" and "database" have been distinct categories. However, in recent years they are getting closer to each other. Sometimes a use-case can be solved with either a materialized view or with a stream processing engine. So they can be seen as alternatives, for some use-cases anyway. 

I'm not sure whether it's important to include this information at all. I just find it to be an interesting and non-obvious realization.

File: blog/2022-04-26-time-series-data-analytics-with-questdb-and-cube.mdx [hunk 3]
```diff
@@ -279,11 +279,11 @@ Next, we can recreate the classic price-over-time graph:
 />
 
 Additionally with Cube, you can pre-aggregate data to speed up the queries. Cube
-will create materialized rollups of specified dimensions and measures
-internally, and use aggregate awareness logic to route the queries. This logic
-will use the most optimal pre-aggregation available to serve the query instead
-of processing the raw dataset. You can define pre-aggregations within your data
-model as shown below.
+will create [materialized](/glossary/materialized-view/) rollups of specified
```
Comment (amyshwang) [inline blog/2022-04-26-time-series-data-analytics-with-questdb-and-cube.mdx:282]: @jerrinot I gather here we're also talking about materialized views so I added a link.

Comment (jerrinot) [reply blog/2022-04-26-time-series-data-analytics-with-questdb-and-cube.mdx:282]: looks good to me

**Review by jerrinot — APPROVED** [review]

**Pull request merged**